### DISCUSSION TITLE
How does the json-file work?

### DISCUSSION THREAD

**Discussion by Appelg** [yozik04/nibe#157] [General]
Python is not my main language. Just reading the code, I cannot figure out how the json-file works.

The s-1255 CSV:
https://github.com/yozik04/nibe/blob/master/nibe/data/s1155_s1255.csv
The json:
https://github.com/yozik04/nibe/blob/master/nibe/data/s1155_s1255.json

From the **CSV**, we can see that "outdoor temp" has register address (1).

The **json** says:
`  "30002": {
    "title": "Current outdoor temperature (BT1)",
    "factor": 10,
    "unit": "\u00b0C",
    "size": "s16",
    "name": "current-outdoor-temperature-bt1-30002"
  },`

Where is the mapping to adress 1 in the json? I dont get it, sorry.

**Comment by yozik04** [comment]
We generate json from CSV. CSV is exported from the heat pump.
Different registry types have different starting address.
What are you trying to solve?

**Reply by Appelg** [reply]
2 things:
1) I want to make sure that a value I extract is actually from the register I expect
2) If I know I want for example the value from register 2166, it would be nice to just  be able to find that in the json.

My proposal is a new field in the json-structure that holds the register-id from the CSV.

**Reply by elupus** [reply]
We should actually change the name or description to include the original identifier. But its a breaking change for home assistant to change the name, so its a bit hard.

**Reply by Appelg** [reply]
Ok! But can you at least link the code that gets the original identifier from a certain JSON id?
Or is the original ID not used when communicating with the Nibe system?

Will another field break Home Assistant?
E.g

"30002": { 
"title": "Current outdoor temperature (BT1)",
 "factor": 10, 
"unit": "\u00b0C", 
"size": "s16", 
"name": "current-outdoor-temperature-bt1-30002",
**"registerId": 1**
},

**Reply by yozik04** [reply]
https://github.com/yozik04/nibe/blob/12802fc7dd65e30e71490dd6e516dc27dd843665/nibe/connection/modbus.py#L29-L38

**Reply by elupus** [reply]
No an extra field would not break home assistant. I dont think changing name does either actually come to think of it. The unique id is based on the parameter number anyway. So we could change the name. Old installs will not reflect it, but i dont think that matters too much.
